### PR TITLE
update etcd

### PR DESCRIPTION
**Pull request by siddontang**
@qiuyesuifeng @huachaohuang @overvenus

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -104,7 +104,12 @@ func NewServer(cfg *Config) (*Server, error) {
 	}
 
 	if err = waitEtcdStart(client, endpoints[0]); err != nil {
-		return nil, errors.Trace(err)
+		// See https://github.com/coreos/etcd/issues/6067
+		// Here may return "not capable" error because we don't start
+		// all etcds in initial_cluster at same time, so here just log
+		// an error.
+		// Note that pd can not work correctly if we don't start all etcds.
+		log.Errorf("etcd start failed, err %v", err)
```
Comment (huachaohuang) [inline server/server.go]: If it can not work, why not return error?

Comment (siddontang) [inline server/server.go]: the server may be started later, and then it can work. 
here we just guarantee pd can start ok.

**Comment by huachaohuang** [comment]
LGTM

File: server/server_test.go [hunk 2]
```diff
@@ -100,9 +100,6 @@ func (s *testLeaderServerSuite) TestLeader(c *C) {
 	svr.Close()
 	delete(s.svrs, leader1.GetAddr())
 
-	// now, another two servers must select a leader
-	s.setUpClient(c)
```
Comment (qiuyesuifeng) [inline server/server_test.go]: Should we remove any other redundant codes?

Comment (siddontang) [inline server/server_test.go]: no, only this is redundant.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**